spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/observation/AppDeployerContext.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2013-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.deployer.spi.app.observation;
+
+import io.micrometer.observation.transport.SenderContext;
+
+import org.springframework.cloud.deployer.spi.app.AppDeployer;
+import org.springframework.cloud.deployer.spi.app.AppScaleRequest;
+import org.springframework.cloud.deployer.spi.core.AppDeploymentRequest;
+import org.springframework.cloud.deployer.spi.core.RuntimeEnvironmentInfo;
+import org.springframework.lang.Nullable;
+
+/**
+ * A {@link SenderContext} for {@link AppDeployer}.
+ *
+ * @author Marcin Grzejszczak
+ */
+public class AppDeployerContext extends SenderContext<Object> {
```

Comment:
Where can I find out more about the carrier (aka `C`) in `SenderContext<C>`? I am curious why its `Object` here.

Reply:
That's because we don't have any real carrier here. We're using this approach with `SenderContext` and the corresponding tracing handlers to set the span kinds and remote service names automatically